src/github.ts

```diff
@@ -1976,6 +1976,29 @@ export class GitHub {
       force: true,
     });
   }
+
+  async getLabels(): Promise<string[]> {
+    const {owner, repo} = this.repository;
+    const labels = await this.request('GET /repos/{owner}/{repo}/labels', {
+      owner,
+      repo,
+    });
+    return labels.data.map(l => l.name);
+  }
+
+  async createLabels(labels: string[]) {
```

Comment:
question: what permissions are needed for this to work?

Reply:
Good question, I need to check...

Reply:
It seems to be both "Issues" and/or "Pull Requests": https://docs.github.com/en/rest/overview/permissions-required-for-fine-grained-personal-access-tokens?apiVersion=2022-11-28#repository-permissions-for-pull-requests. So we should be good.

Reply:
Great! Can you double check with a test repo just to make sure? (I'm fairly paranoid about GitHub doing weird things)

Reply:
You're right, I will double check, I don't want to break the whole CI

Reply:
I can confirm, that seems to work with only Pull Request or only Issues as writeable :)